pkg/resourcemanager/disk/client.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package disk
+
+import (
+	"context"
+	"strings"
+
+	compute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/config"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/iam"
+	"github.com/Azure/azure-service-operator/pkg/secrets"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type AzureDiskClient struct {
+	SecretClient secrets.SecretClient
+	Scheme       *runtime.Scheme
+}
+
+func NewAzureDiskClient(secretclient secrets.SecretClient, scheme *runtime.Scheme) *AzureDiskClient {
+	return &AzureDiskClient{
+		SecretClient: secretclient,
+		Scheme:       scheme,
+	}
+}
+
+func getDiskClient() compute.DisksClient {
+	computeClient := compute.NewDisksClientWithBaseURI(config.BaseURI(), config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
+	computeClient.Authorizer = a
+	computeClient.AddToUserAgent(config.UserAgent())
+	return computeClient
+}
+
+func (m *AzureDiskClient) CreateDisk(ctx context.Context, location string, resourceGroupName string, resourceName string, createOption string, diskSizeGB int32) (future compute.DisksCreateOrUpdateFuture, err error) {
+
+	client := getDiskClient()
+
+	diskCreateOption := compute.DiskCreateOption(strings.TrimSpace(createOption))
+
+	// Initialize Disk API Spec properties.
+	diskAPISpec := compute.Disk{
+		Location: &location,
+		DiskProperties: &compute.DiskProperties{
+			CreationData: &compute.CreationData{
+				CreateOption: diskCreateOption,
+			},
+			DiskSizeGB: &diskSizeGB,
+		},
+	}
+
+	future, err = client.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		resourceName,
+		diskAPISpec,
+	)
+
+	return future, err
```

Comment:
Our pattern has been to force the future to complete even if it isn't done yet and then return the result of that call from the client. Like this:
```
	result, err := future.Result(redisClient)
	return &result, err
```